software/contrib/menu.md

```diff
@@ -46,6 +46,66 @@ one to import your class and one to add the class to the list. Use the existing
 
 Note that the scripts are sorted before being displayed, so order in this file doesn't matter.
 
+## Save/Load Script State
+
+You can add a bit of code to enable your script to save state upon change, and load previous state at startup.
+
+When adding save state functionality to your script, there are a few important considerations to keep in mind:
+
+1. Frequency of saves - scripts should only save state to disk when state changes, and should not save too frequently because os write operations are expensive in terms of time. Saving too frequently will affect the performance of a script.
+1. Save state file size - The pico only has about 1MB of free space available so save state storage format is important to keep as minimal as possible.
+1. No externally influenced input - The instance variables your script saves should not be externally influenced, meaning you should not save the current knob position, current analog input value or current digital input value.
+
+Here is an extension of the script above with some added trivial features that incorporate saveing and loading script state.
+
+```python
+from europi import oled
+from europi_script import EuroPiScript
+
+class HelloWorld(EuroPiScript):
+    def __init__(self):
+        state = super().load_state_json()  # 1
+
+        self.counter = state.get("counter", 0)  # 2
+        self.enabled = state.get("enabled", True)
+
+        @din.handler
+        def increment_counter():
+            if self.enabled:
+                self.counter += 1
+                self.save_state()  # 3
+        
+        @b1.handler
+        def toggle_enablement():
+            self.enabled = not self.enabled
+            self.save_state()  # 3
+
+    def save_state(self):  # 4
```

Comment:
Is the intention here to override `EuroPiScript.save_state()`? I don't think that we should recommend this naming, it's going to lead to confusion.